frontend/js/src/settings/link-listens/LinkListens.tsx

```diff
@@ -100,6 +102,41 @@ export default function LinkListensPage() {
   const [unlinkedListens, setUnlinkedListens] = React.useState<
     Array<UnlinkedListens>
   >(unlinkedListensProps);
+  const [filterType, setFilterType] = React.useState("album"); // Default filter type
+  const [searchQuery, setSearchQuery] = React.useState(""); // To store the search input
+  const handleSearch = () => {
+    if (filterType === "album") {
+      const fuzzysearch = new Fuse(unlinkedListensProps, {
+        keys: ["release_name"],
+      });
+      const filtered = fuzzysearch
+        .search(searchQuery)
+        .map((result) => result.item);
+      setUnlinkedListens(filtered);
```

Comment:
In entity search, users should be redirected to the first page of results whenever a new search query is initiated.